examples/AspNetCore/Instrumentation.cs

```diff
@@ -0,0 +1,50 @@
+// <copyright file="Instrumentation.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace Examples.AspNetCore;
+
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+
+/// <summary>
+/// It is recommended to use a custom type to hold references for
+/// ActivitySource and Instruments. This avoids possible type collisions
+/// with other components in the DI container.
+/// </summary>
+public class Instrumentation : IDisposable
```

Comment:
Maybe (otherwise it would require virtual Dispose?):

```suggestion
public sealed class Instrumentation : IDisposable
```

Reply:
Never mind, I just saw how the namespaces are organized. Closing.